File: 2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.rb
# @param {String} s
# @param {Integer} k
# @param {Character} fill
# @return {String[]}
def divide_string(s, k, fill)
    result = []

    (0...s.length).step(k) do |i|
        group = s[i, k]
        group = group.ljust(k, fill) if group.length < k
        result << group
    end

    result
end

File: 2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.go
func divideString(s string, k int, fill byte) []string {
	var result []string
	for i := 0; i < len(s); i += k {
		end := i + k
		if end > len(s) {
			end = len(s)
		}
		group := s[i:end]

		if len(group) < k {
			for len(group) < k {
				group += string(fill)
			}
		}

		result = append(result, group)
	}
	return result
}